### PR TITLE
test: Add support for injecting Spartacus TestConfig chunk from ssr testing framework

### PR DESCRIPTION
**Pull request by Zeyber**
We're introducing `TestConfigServerModule`, which reads a cookie from the `REQUEST` (but not `DOCUMENT.cookie` - which is not implemented in runtime of the `Domino` lib). 

Also added a convenient new API to our SSR e2e testing framework for passing the extra `testConfig`.

Closes: [CXSPA-6633](https://jira.tools.sap/browse/CXSPA-6633)

### CODE REVIEW

File: projects/ssr-tests/tsconfig.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
+  "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "baseUrl": "./",
```
Comment (Platonn) [inline projects/ssr-tests/tsconfig.json:3]: Note:
- inheriting baseUrl being root dir - so all `Config` interface augmentations are taken into account (for type safety of `TestConfig`
- inheriting `paths` mapping for all libs (.e.g. `@sparatcus/core` resolved to `projects/core/src/public_api`) - so we can import `TestConfig` from `@spartacus/core`

File: core-libs/setup/ssr/testing/test-config-server.module.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  return (result && decodeURIComponent(result[1])) || '';
+}
+
+export function parseConfigJSON(config: string) {
+  try {
+    return JSON.parse(decodeURIComponent(config));
+  } catch (_) {
+    return {};
+  }
+}
```
Comment (pawelfras) [inline core-libs/setup/ssr/testing/test-config-server.module.ts:24]: `query:` Are these functions repeated here to not expose them to `@spartacus/core` public API?

**Review by pawelfras — DISMISSED** [review]
LGTM

File: core-libs/setup/ssr/testing/test-config-server.module.ts (hunk 2 above)
Comment (Platonn) [reply core-libs/setup/ssr/testing/test-config-server.module.ts:24]: yes

**Review by pawelfras — DISMISSED** [review]

**Review by pawelfras — APPROVED** [review]

**Pull request merged**